datastore/tests/unit/test_helpers.py

```diff
@@ -135,6 +135,27 @@ def test_mismatched_value_indexed(self):
         with self.assertRaises(ValueError):
             self._call_fut(entity_pb)
 
+    def test_index_mismatch_ignores_empty_list(self):
+        from google.cloud.proto.datastore.v1 import entity_pb2
+        from google.cloud.datastore.helpers import _new_value_pb
+
+        _PROJECT = 'PROJECT'
+        _KIND = 'KIND'
+        _ID = 1234
+        entity_pb = entity_pb2.Entity()
+        entity_pb.key.partition_id.project_id = _PROJECT
+        entity_pb.key.path.add(kind=_KIND, id=_ID)
+
+        array_val_pb = _new_value_pb(entity_pb, 'baz')
+        array_pb = array_val_pb.array_value.values
+
+        unindexed_value_pb1 = array_pb.add()
+        unindexed_value_pb1.integer_value = 10
```

Comment:
This new test doesn't exercise the bug:  it passes even if I back out the change to `helpers`.

Reply:
We should be leaving the list *empty*, and verifying that it propagates the `exclude_from_indexes`.